Makefile

```diff
@@ -50,15 +50,32 @@ else
 	stack install --pedantic --test --bench --no-run-benchmarks --local-bin-path=dist
 endif
 
-.PHONY: delete-cache-on-linker-errors
-delete-cache-on-linker-errors:
+# Clean
+.PHONY: clean
+clean: delete-cache-on-linker-errors
 	rm -rf ~/.cache/hie-bios
 ifdef CABAL_DIR
 	rm -rf $(CABAL_DIR)/store
 else
 	rm -rf ~/.cabal/store
 endif
+ifeq ($(WIRE_BUILD_WITH_CABAL), 1)
 	cabal clean
+else
+	stack clean
+endif
+	$(MAKE) -C services/nginz clean
+	-rm -rf dist
+	-rm -f .metadata
+
+.PHONY: clean-hint
+clean-hint:
+	@echo -e "\n\n\n>>> PSA: if you get errors that are hard to explain,"
+	@echo -e ">>> try 'make clean' and run your command again."
+	@echo -e ">>> see https://github.com/wireapp/wire-server/blob/develop/docs/developer/building.md#linker-errors-while-compiling\n\n\n"
+
+.PHONY: delete-cache-on-linker-errors
+delete-cache-on-linker-errors: clean
```

Comment:
Wouldn't it be better to remove this target and replace it's occurrences in any docs with `clean`?

(`delete-cache-on-linker-errors` is pretty new. So, there shouldn't be much that depends on it. And, IMHO the name is a bit misleading.)

Reply:
ok, i'll do that.

Reply:
9df39b14ae41226cbceeb36a00fa2247ec47423d